src/vos/vos_dtx.c

```diff
@@ -2988,16 +2988,16 @@ vos_dtx_aggregate(daos_handle_t coh)
 		d_iov_set(&kiov, &dce_df->dce_xid, sizeof(dce_df->dce_xid));
 		rc = dbtree_delete(cont->vc_dtx_committed_hdl, BTR_PROBE_EQ,
 				   &kiov, NULL);
-		if (rc != 0 && rc != -DER_NONEXIST) {
+		if (rc == 0) {
+			count++;
```

Comment:
It's better to have an assert in this function when subtracting 'count' for 'vc_dtx_committed_count' and 'vp_dtx_committed_count'.